lib/march_hare/exception_handler/rubify_java_exception_handler.rb

```diff
@@ -0,0 +1,31 @@
+require 'set'
+module MarchHare
+  module RubifyJavaExceptionHandler
+    def initialize(logger)
+      super()
+      @logger = logger
+    end
+
+    def log(msg, error)
+      if is_socket_closed_or_connection_reset(error)
+        logger.warn(msg)
+        logger.warn(error)
+      else
+        logger.error(msg)
+        logger.error(error)
+      end
+    end
+
+    attr_reader :logger
+
+    SOCKET_CLOSE_OR_CONNECTION_RESET_MESSAGE = Set.new([
+      "Connection reset",
+      "Socket closed",
+      "Connection reset by peer"
+    ])
+
+    def is_socket_closed_or_connection_reset(error)
```

Comment:
I understand the intent here but lumping these together makes little sense. "TCP connection reset" means a TCP connection was refused by the peer. Socket closure is a different scenario.

Naming predicate methods as `is_predicate` in Ruby is very uncommon. Use `predicate?` instead.

I'd rename this to `warning?` or similar.

Reply:
Yeah good idea regarding the naming I just took the exact behavior & naming from the private java method. See https://github.com/rabbitmq/rabbitmq-java-client/blob/master/src/main/java/com/rabbitmq/client/impl/ForgivingExceptionHandler.java#L112-L131

Reply:
I guess we could also decide to simply not do this separation between certain exceptions and just always log it as error

Reply:
Good idea 👍